bpo-44940: Clarify the documentation of re.findall() (#27849)
python/cpython: Doc/library/re.rst

.. function:: findall(pattern, string, flags=0)

   Return all non-overlapping matches of *pattern* in *string*, as a list of
   strings or tuples.  The *string* is scanned left-to-right, and matches
   are returned in the order found.  Empty matches are included in the result.

   The result depends on the number of capturing groups in the pattern.
   If there are no groups, return a list of strings matching the whole
   pattern.  If there is exactly one group, return a list of strings

>    matching the group.  If more than one groups are present, return a list

Reviewer: Ditto to what vedgar said above about `that`. These are just minor suggestions, feel free to ignore.
```suggestion
   matching that group.  If multiple groups are present, return a list
```

   of tuples of strings matching the groups.  Non-capturing groups do not

>    affect the result.

Author: ```suggestion
   affect the form of the result.
```

      >>> re.findall(r'\bf[a-z]*', 'which foot or hand fell fastest')
      ['foot', 'fell', 'fastest']

>       >>> re.findall(r'(\w+)=(\d+)', 'set width=20 and height=10')

Reviewer: Great example! The sentence above was a little hard to parse. But this example clears everything up.
